src/vm/page.c

```diff
@@ -245,24 +245,40 @@ void _page_showSizes(void)
 }
 
 
+static unsigned int page_digits(unsigned int n, unsigned int base)
+{
+	return (n < base) ? 1 : 1 + page_digits(n / base, base);
+}
+
+
 void _page_showPages(void)
 {
 	addr_t a;
 	page_t *p;
-	unsigned int rep, i, k;
+	unsigned int rep, i, k, w = 4;
 	char c;
 
+	lib_printf("vm: ");
 	for (i = 0, a = 0; i < (pages.freesz + pages.allocsz) / SIZE_PAGE; i++) {
-
 		p = &pages.pages[i];
 
 		/* Print markers in case of memory gap */
 		if (p->addr > a) {
-			if ((p->addr - a) / SIZE_PAGE >= 4)
-				lib_printf("[%dx]", (p->addr - a) / SIZE_PAGE);
+			if ((rep = (p->addr - a) / SIZE_PAGE) >= 4) {
+				if ((w += page_digits(rep, 10) + 3) > 80) {
+					lib_printf("\nvm: ");
+					w = page_digits(rep, 10) + 7;
```

Comment:
It would be benefitial to cache `page_digits(rep, 10)` result in a variable.

Reply:
Fixed.

Reply:
What about this PR?